scripts/make_slack_notification_message.py

```diff
@@ -71,20 +85,70 @@ def _format_summary(summary: dict) -> list[dict]:
     if file_changes := summary["file_changes"]:
         abridged_changes = defaultdict(list)
         for change in file_changes:
-            abridged_changes[change["diff_type"]].append(change["name"])
+            abridged_changes[change["diff_type"]].append(
+                # Convert the size diff to MB for speed of assessment, and round
+                {change["name"]: round(change["size_diff"] * (2**-20), 4)}
+            )
         changes = f"```\n{json.dumps(abridged_changes, indent=2)}\n```"
     else:
         changes = "No changes."
 
     return _format_message(url=url, name=name, content=changes)
 
 
-def main(summary_files: list[Path]) -> None:
-    """Format summary files for Slack perusal."""
+def _format_errors(log: str) -> str:
+    """Take a log file from a failed run and return the exception."""
+    failure = log.splitlines()[-1]
+    # We already capture archive validation failures elsewhere, so ignore these.
+    if failure == "RuntimeError: Error: archive validation tests failed.":
+        return None
+    failure = f"```\n{failure}\n```"  # Format as code
+
+    name_re = re.search(
+        r"(?:catalystcoop.pudl_archiver.archivers.classes:155 Archiving )([a-z0-9]*)",
+        log,
+    )
+    name = name_re.group(1)
+
+    url_re = re.search(
+        r"(?:INFO:catalystcoop.pudl_archiver.depositors.zenodo.depositor:PUT )(https:\/\/[a-z0-9\/.]*)",
+        log,
+    )
+    url = url_re.group(1)
+
+    return _format_message(url=url, name=name, content=failure)
+
+
+def _load_summaries(summary_files: list[Path]) -> list[dict]:
     summaries = []
     for summary_file in summary_files:
-        with summary_file.open() as f:
-            summaries.extend(json.loads(f.read()))
+        if summary_file.exists():  # Handle case where no files are found
+            with summary_file.open() as f:
+                summaries.extend(json.loads(f.read()))
+    return summaries
+
+
+def _load_errors(error_files: list[Path]) -> list[str]:
+    errors = []
+    for error_file in error_files:
+        if (
+            error_file.exists() and error_file.stat().st_size > 0
+        ):  # Handle case where no files are found or file is empty
+            with error_file.open() as f:
+                errors.append(f.read())
+    return errors
+
+
+def main(summary_files: list[Path], error_files: list[Path]) -> None:
+    """Format summary files for Slack perusal."""
+    summaries = _load_summaries(summary_files)
+    errors = _load_errors(error_files)
+
+    error_blocks = list(
+        itertools.chain.from_iterable(
+            _format_errors(e) for e in errors if _format_errors(e) is not None
```

Comment:
Rather than calling `_format_errors` twice, consider

```suggestion
            filter(None, (_format_errors(e) for e in errors))
```